curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/handle-a-fulfilled-promise-with-then.english.md

```diff
@@ -0,0 +1,94 @@
+---
+id: 5cdafbd72913098997531681
+title: Handle a Fulfilled Promise with then
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+Promises are most useful when you have a process that takes an unknown amount of time in your code, something asynchronous. Often, a server request. When you make a server request it takes some amount of time, and after it completes you usually want to do something with the response from the server. Introducing the <code>then</code> method. The <code>then</code> method is executed immediately after your promise is fulfilled with <code>resolve</code>. Here’s an example:
+
+```js
+myPromise.then(result => {
+  // do something with the result.
+});
+```
+
+<code>result</code> comes from the argument given to the <code>resolve</code> method.
+</section>
+
+## Instructions
+<section id='instructions'>
+Use the <code>then</code> method on your promise. Log its argument to the console.
+</section>
+
+## Tests
+<section id='tests'>
+
+```yml
+tests:
+  - text: You should properly use the <code>then</code> method.
+    testString: assert(code.match(/(makeServerRequest|\))\.then\(\s*(\s*function\s*\(\s*result\s*\)\s*|\s*result\s*=>\s*|\s*\(\s*result\s*\)\s*=>\s*){[^}]*}\s*\)/g));
+  - text: You should log the <code>then</code> argument (<code>result</code>) to the console.
+    testString: assert(code.match(/{[^c]*console\.log\(\s*result\s*\)[^}]*}/g));
+```
+
+</section>
+
+## Challenge Seed
+<section id='challengeSeed'>
+<div id='js-seed'>
+
+```js
+const makeServerRequest = new Promise((resolve, reject) => {
+  const responseFromServer = true;
+	
+  if(responseFromServer) {
+    resolve("We got the data");
+  } else {	
+    reject("Data not received");
+  }
+});
+```
+
+</div>
+
+## Challenge Seed
+<section id='challengeSeed'>
+<div id='js-seed'>
+
+```js
+const makeServerRequest = new Promise((resolve, reject) => {
+  const responseFromServer = true;
```

Comment:
```suggestion
  let responseFromServer;
  \\ the server responds and responseFromServer is set to true or false
```
Mirroring the change to the other challenge.

Reply:
For the .then and .catch lessons - if it isn't given a true or false value, resolve or reject won't be executed. Therefore, .then and .catch won't be executed. So I'm thinking something like this... 
```suggestion
\\ responseFromServer is set to true to represent a successful response from a server
let responseFromServer = true;
```

Reply:
Yeah, that works.  I didn't like `const thing = true` followed by `if(thing)`, but the way you have it now gets across what's being abstracted away.